dartagnan/src/main/java/com/dat3m/dartagnan/expression/utils/ExpressionHelper.java

```diff
@@ -0,0 +1,14 @@
+package com.dat3m.dartagnan.expression.utils;
+
+import com.dat3m.dartagnan.expression.Expression;
+import com.google.common.base.Preconditions;
+
+public class ExpressionHelper {
+
+    private ExpressionHelper() {}
+
+    public static void checkSameType(Expression x, Expression y) {
+        Preconditions.checkArgument(x.getType().equals(y.getType()),
+                "The types of %s and %s do not match.", x, y);
+    }
+}
```

Comment:
I think there is code touched by this PR that can use this method and still doesn't

Reply:
Probably. I will do a pass over the whole code to cleanup more things.